### PR TITLE
Trigger silent notice when calling a deprecated mocked method

### PR DESCRIPTION
**Pull request by nicolas-grekas**
This is #276 on master

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Fixes #271

File: src/Framework/MockObject/Generator.php [hunk 1]
```diff
@@ -971,9 +978,10 @@ private function generateMockedMethodDefinitionFromExisting($templateDir, Reflec
      * @param  string $reference
      * @param  bool   $callOriginalMethods
      * @param  bool   $static
+     * @param  string $deprecation
```
Comment (vincentchalamon) [inline src/Framework/MockObject/Generator.php]: Should be:
```php
/**
 * @param string|bool $deprecation
 */
```

**Comment by vincentchalamon** [comment]
:+1:

**Comment by sebastianbergmann** [comment]
Merged manually, thanks.

File: src/Framework/MockObject/Generator.php [hunk 2]
```diff
@@ -944,6 +944,12 @@ private function generateMockedMethodDefinitionFromExisting($templateDir, Reflec
             $returnType = '';
         }
 
+        if (preg_match('#\*[ \t]*+@deprecated[ \t]*+(.*?)\r?+\n[ \t]*+\*(?:[ \t]*+@|/$)#s', $method->getDocComment(), $deprecation)) {
```
Comment (sebastianbergmann) [inline src/Framework/MockObject/Generator.php]: I think the notice should also be triggered when there is a `@deprecated` in `$method->getDeclaringClass()->getDocComment()`.

**Pull request closed without merging**